presto-main/src/main/java/com/facebook/presto/operator/scalar/MapConcatFunction.java

```diff
@@ -0,0 +1,100 @@
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.metadata.FunctionInfo;
+import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.metadata.ParametricScalar;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.block.InterleavedBlockBuilder;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.spi.type.TypeSignature;
+import com.facebook.presto.type.MapType;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Map;
+
+import static com.facebook.presto.metadata.Signature.comparableTypeParameter;
+import static com.facebook.presto.type.TypeUtils.parameterizedTypeName;
+import static com.facebook.presto.util.Reflection.methodHandle;
+
+public class MapConcatFunction
+        extends ParametricScalar
+{
+    public static final MapConcatFunction MAP_CONCAT_FUNCTION = new MapConcatFunction();
+    private static final String FUNCTION_NAME = "map_concat";
+    private static final Signature SIGNATURE = new Signature(FUNCTION_NAME, ImmutableList.of(comparableTypeParameter("K"), comparableTypeParameter("V")), "map<K,V>", ImmutableList.of("map<K,V>", "map<K,V>"), false, false);
+    private static final MethodHandle METHOD_HANDLE = methodHandle(MapConcatFunction.class, FUNCTION_NAME, Type.class, Block.class, Block.class);
+
+    @Override
+    public Signature getSignature()
+    {
+        return SIGNATURE;
+    }
+
+    @Override
+    public boolean isHidden()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isDeterministic()
+    {
+        return true;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Flattens a given array";
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type keyType = types.get("K");
+        Type valueType = types.get("V");
+        Type mapType = typeManager.getParameterizedType(StandardTypes.MAP, ImmutableList.of(keyType.getTypeSignature(), valueType.getTypeSignature()), ImmutableList.of());
+        MethodHandle methodHandle = METHOD_HANDLE.bindTo(mapType);
+        TypeSignature typeSignature = parameterizedTypeName("map", keyType.getTypeSignature(), valueType.getTypeSignature());
+        Signature signature = new Signature(FUNCTION_NAME, typeSignature, typeSignature, typeSignature);
+        return new FunctionInfo(signature, getDescription(), isHidden(), methodHandle, isDeterministic(), false, ImmutableList.of(false, false));
+    }
+
+    public static Block map_concat(Type mapType, Block leftMap, Block rightMap)
```

Comment:
`public static Block map_concat(Type keyType, Type valueType, Block leftMap, Block rightMap)`
Change the implementation accordingly.